src/Middleware/ResponseCompression/src/BodyWrapperStream.cs

```diff
@@ -65,14 +68,29 @@ public override long Position
             set { throw new NotSupportedException(); }
         }
 
+        public Stream Body => this;
+
+        public PipeWriter Writer
+        {
+            get
+            {
+                if (_pipeAdapter == null)
+                {
+                    _pipeAdapter = PipeWriter.Create(Body, new StreamPipeWriterOptions(leaveOpen: true));
```

Comment:
@davidfowl I wish we made leaveOpen: true the default.

Reply:
Yea, maybe it should be, though other BCL apis it's true by default.